pkg/kubelet/lifecycle/handlers.go

```diff
@@ -92,15 +92,19 @@ func resolvePort(portReference intstr.IntOrString, container *api.Container) (in
 		return portReference.IntValue(), nil
 	}
 	portName := portReference.StrVal
-	port, err := strconv.Atoi(portName)
-	if err == nil {
-		return port, nil
-	}
-	for _, portSpec := range container.Ports {
-		if portSpec.Name == portName {
-			return int(portSpec.ContainerPort), nil
+
+	if len(portName) > 0 {
```

Comment:
Not sure if this is a good idea. The behavior changes drastically. if the portName is "" earlier we would look in container.ports (as err returned from Atoi is not nil). With this change we will drop down to the end of the method and won't look in container.Ports